### ISSUE TITLE
IPAM for iTop upgrade from 2.5.1 to 2.7.1.

### ISSUE THREAD

**Issue by he76ko** [TeemIp/teemip-core-ip-mgmt#12]
Upgrade unsuccessfull on a shared database where itop database uses 'itop' prefix.

```
Failed to issue SQL query: query = update priv_backgroundtask set next_run_date= '2020-01-01 00:00:00' where class_name = 'ReleaseIPsFromObsoleteCIs', mysql_errno = 1146, mysql_error = Table 'itop.priv_backgroundtask' doesn't exist | SetupLog
2021-11-11 15:44:19 | Error   |       | Debug trace: #0 /var/www/itop.deverto/web/core/cmdbsource.class.inc.php(672): CMDBSource::DBQuery('update priv_bac...')
#1 /var/www/itop.deverto/web/setup/extensionsmap.class.inc.php(455) : eval()'d code(122): CMDBSource::Query('update priv_bac...')
#2 /var/www/itop.deverto/web/setup/runtimeenv.class.inc.php(1100): IPManagementInstaller::AfterDatabaseCreation(Object(Config), '2.5.1', '2.7.1')
#3 /var/www/itop.deverto/web/env-production/itop-hub-connector/ajax.php(334): RunTimeEnvironment->CallInstallerHandlers(Array, Array, 'AfterDatabaseCr...')
#4 /var/www/itop.deverto/web/pages/exec.php(62): require_once('/var/www/itop.d...')
#5 {main} | SetupLog
2021-11-11 15:44:19 | Ok      |       | <---- Exiting read only mode | SetupLog
```

```
mysql> describe itop.itoppriv_backgroundtask;
+----------------------+-------------------------+------+-----+---------+----------------+
| Field                | Type                    | Null | Key | Default | Extra          |
+----------------------+-------------------------+------+-----+---------+----------------+
| id                   | int(11)                 | NO   | PRI | NULL    | auto_increment |
| class_name           | varchar(255)            | YES  | MUL |         |                |
| first_run_date       | datetime                | YES  |     | NULL    |                |
| latest_run_date      | datetime                | YES  |     | NULL    |                |
| next_run_date        | datetime                | YES  |     | NULL    |                |
| total_exec_count     | int(11)                 | YES  |     | 0       |                |
| latest_run_duration  | decimal(8,3)            | YES  |     | 0.000   |                |
| min_run_duration     | decimal(8,3)            | YES  |     | 0.000   |                |
| max_run_duration     | decimal(8,3)            | YES  |     | 0.000   |                |
| average_run_duration | decimal(8,3)            | YES  |     | 0.000   |                |
| running              | tinyint(1)              | YES  |     | 0       |                |
| status               | enum('active','paused') | YES  |     | active  |                |
| system_user          | varchar(255)            | YES  |     |         |                |
+----------------------+-------------------------+------+-----+---------+----------------+
```

**Comment by xtophe38** [comment]
I'll upgrade the code so that the DB prefix is taken into account. In the meantime, you can hardcode your prefix in the IPManagementInstaller declared in the module.teemip-ip-mgmt.php file.

**Comment by xtophe38** [comment]
Upgrade is no available in 3.0.0 release.
